### DISCUSSION TITLE
Palettized blitting?

### DISCUSSION THREAD

**Discussion by TheRealQuantam** [python-pillow/Pillow#6527] [Q&A] [answered]
Is there a good way of blitting palettized Pillow Images? Specifically, this requires the ability to paste a palettized image with some kind of transparency (either a transparent palette entry or an RGBA palette with binary A values) at a specified place on another image with the same palette? Bonus points if you can also paste a section (rather than the whole) of the source image.

**Comment by radarhere** [comment, marked as answer]
It sounds to me like you're just describing [`paste()`](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.paste) with `box` and `mask` arguments, and a [`crop()`](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.crop) operation.

Using https://github.com/python-pillow/Pillow/blob/main/Tests/images/transparent.png,
```python
from PIL import Image
im = Image.open("transparent.gif")
im2 = im.crop((0, 0, 100, 100))
im.paste(im2, (50, 50), im2.convert("LA"))
im.save("out.png")
```
gives
![out](https://user-images.githubusercontent.com/3112309/186159193-989f5d85-ac17-489c-b253-ed133f1ef892.png)

**Reply by radarhere** [reply]
@TheRealQuantam did that answer your question?

**Reply by TheRealQuantam** [reply]
I never did reply to this, did I. Yes, this works for the use case described. I had been hoping there was a way to make it use the alpha channel or something, but for my purposes storing the image in PA and passing the alpha channel as mask works fine, and is about twice as fast as numpy (what I've been using to assemble my images before displaying/writing them until now).
